csrc/executor.h

```diff
@@ -179,26 +179,89 @@ class FusionExecutor : public NonCopyable {
     return measure_kernel_time_ ? kernel_time_ms_ : 0;
   }
 
+  //! Returns the input bytes accessed for a kernel
+  //! \note It is important to sample the args struct prior to adding the
+  // 1    output to the args struct
+  int64_t inputBytesProcessed(KernelArgumentHolder& args) {
+    int64_t total_bytes = 0;
+    if (!bytes_processed_per_input_.has_value()) {
+      int64_t num_bytes = 0;
+      bytes_processed_per_input_ = std::optional<std::vector<int64_t>>(
+          std::vector<int64_t>(args.size(), 0));
+      // Figure how many bytes are inputs, outputs, and temporary buffers
+      for (auto i : c10::irange(args.size())) {
+        if (args[i]->is<at::Tensor>()) {
+          auto t = args[i]->as<at::Tensor>();
+          num_bytes = t.numel() *
```

Comment:
I'm not sure this is the right measure for non-packed tensors. How about `t.storage().nbytes()`?

Reply:
second this. numel isn't right for expanded tensors e.t.c. storage is also deprecated btw. so `t.untyped_storage().nbytes()`

Reply:
Thanks guys. I wrote those lines originally, but I didn't know how to get the actual storage.

Reply:
`at::Tensor` does not appear to have the method `untyped_storage()` in the C++ API.  It does exist in the Python API.